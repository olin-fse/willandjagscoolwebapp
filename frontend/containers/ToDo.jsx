import React, { Component } from 'react'

import List from '../components/List.jsx'
import Input from '../components/Input.jsx'
import Title from '../components/Title.jsx'

class ToDo extends React.Component {
  constructor(props){
    super(props);
  }

  state = {
    todos: [null],
  }

  componentDidMount(){
    console.log("i mounted boiii");

    fetch('/showTodos', {
      method: 'GET',
      headers: {
        "Content-type": "application/json; charset=UTF-8"
      },
    })
      .then((response) => { return response.json(); })
      .then((json) => { console.log(json); this.setState(json); });
  }

  onAddTodo = (text) => {
    const {todos} = this.state

    const data = {
      userId: this.props.user.Id,
      text: text,
    }

    console.log(JSON.stringify(data));

    fetch('/addTodo', {
      method: 'POST',
      headers: {
        "Content-type": "application/json; charset=UTF-8"
      },
      body: JSON.stringify(data)
    })
      .then((response) => { return response.json(); })
      .then((json) => { 
        this.setState({
          todos: [text, ...todos],
        })
      });
  }

  onRemoveTodo = (index) => {
    const {todos} = this.state

    var deletedItem = todos[index];

    console.log(JSON.stringify({deletedItem}))

    fetch('/deleteTodo', {
      method: 'POST',
      headers: {
        "Content-type": "application/json; charset=UTF-8"
      },
      body: JSON.stringify({deletedItem})
    })
      .then((response) => { console.log(response); return response.json(); })
      .then((json) => { 
        this.setState({
          todos: todos.filter((todo, i) => i !== index),
        })
      });
  }

  render() {
    const {todos} = this.state

    return (
      <div style={styles.container}>
        <Title>
          To-Do List
        </Title>
        <Input
          placeholder={'Type a todo, then hit enter!'}
          onSubmitEditing={this.onAddTodo}
        />
        <List
          list={todos}
          onClickItem={this.onRemoveTodo}
        />
      </div>
    )
  }
}

const styles = {
  container: {
    display: 'flex',
    flexDirection: 'column',
  }
}

export default ToDo;