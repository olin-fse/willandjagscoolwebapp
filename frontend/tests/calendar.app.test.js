import React from 'react';
import ToDo from '../containers/ToDo.jsx';
import renderer from 'react-test-renderer';

test('Entry is removed when clicked', () => {
    const component = renderer.create(
        <ToDo/>,
      );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    //remove first event in to-do list
    tree.props.onRemoveTodo(0);
    //re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});