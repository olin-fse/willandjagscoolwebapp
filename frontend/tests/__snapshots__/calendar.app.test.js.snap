// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry is removed when clicked 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "skyblue",
        "padding": 15,
      }
    }
  >
    <div
      style={
        Object {
          "color": "white",
          "textAlign": "center",
        }
      }
    >
      To-Do List
    </div>
  </div>
  <input
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Type a todo, then hit enter!"
    style={
      Object {
        "borderWidth": 0,
        "fontSize": "100%",
        "padding": 15,
      }
    }
    type="text"
    value=""
  />
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
      }
    }
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "whitesmoke",
          "marginBottom": 5,
          "padding": 15,
        }
      }
    >
      Click to remove
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "whitesmoke",
          "marginBottom": 5,
          "padding": 15,
        }
      }
    >
      Learn React
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "whitesmoke",
          "marginBottom": 5,
          "padding": 15,
        }
      }
    >
      Write Code
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "whitesmoke",
          "marginBottom": 5,
          "padding": 15,
        }
      }
    >
      Ship App
    </div>
  </div>
</div>
`;
